my-sms/db/migrate/20210212025754_move_student_title_str_to_title_model.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class MoveStudentTitleStrToTitleModel < ActiveRecord::Migration
+  def up
+    # update students table
+    change_table :students do |t|
+      t.rename(:title, :title_old)
+      t.references :title
+    end
+
+    logger.warn 'Run rake db:migrate:migrate_students to convert existing title Strings in title_old column.'
+  end
+
+  def down
+    # re-create string title column
+    change_table :students do |t|
+      t.string :title_str
+    end
+
+    # undo title model association without assuming title implementation in model
+    Student.find_each do |student|
+      student.update_attribute(:title_str, Title.find(student[:title_id]).name)
```

Comment:
Nitpick: could simplify with `student.title.name`

Reply:
You don't have to change this now, but it's worth knowing that you can also do `Student.includes(:title).find_each do ...` to avoid the n+1 query problem, but we might cover this a bit later

Reply:
Was wanting to reduce model dependence by doing it this way.